### PR TITLE
Release process updations

### PR DESCRIPTION
**Pull request by cijothomas**
Fixes #.

## Changes

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: build/RELEASING.md [hunk 1]
```diff
@@ -44,49 +39,57 @@ Only for Maintainers.
...
+
+   ```sh
+   git tag -a 1.0.0-rc2 -m "1.0.0-RC2"
+   git push origin 1.0.0-rc2
+   ```
+
+We use [MinVer](https://github.com/adamralph/minver) to do versioning,
+which produces version numbers based on git tags.
+
+6.Open [Pack and publish to MyGet
```
Comment (cijothomas) [inline build/RELEASING.md:75]: The key change is, we tag and then run release workflow. So the package will be named exactly as the tag name. i.e if tag was 1.0.0, version would be 1.0.0.
Then the daily myget publish will automatically do, 1.0.1-alpha.1 series.
When we release 1.0.1, we tag 1.0.1 and run release workflow which push 1.0.1 nugets.
Then daily myget publish starts 1.0.2-alpha.1 series.
(all of which is easily customizable by pushing required tags)

File: build/RELEASING.md [hunk 2]
```diff
@@ -44,49 +39,57 @@ Only for Maintainers.
...
     }
 ```
 
 4.Submit PR with the above changes, and get it merged.
 
-5.Open [Pack and publish to MyGet
+5.Tag Git with version to be released e.g.:
+
+   ```sh
+   git tag -a 1.0.0-rc2 -m "1.0.0-RC2"
```
Comment (reyang) [inline build/RELEASING.md:68]:
```suggestion
   git tag -a 1.0.0-rc2 -m "1.0.0-rc2"
```

File: build/RELEASING.md [hunk 3]
```diff
@@ -44,49 +39,57 @@ Only for Maintainers.
...
+
+6.Open [Pack and publish to MyGet
    workflow](https://github.com/open-telemetry/opentelemetry-dotnet/actions?query=workflow%3A%22Pack+and+publish+to+Myget%22)
    and manually trigger a build. At the end of this, MyGet will have the
-   packages. The package name will be the tag name used in step1 appended with
-   ".1".
+   packages. The package name will be the tag name used in step 5.
 
-6.Validate using MyGet packages. Basic sanity checks :)
+7.Validate using MyGet packages. Basic sanity checks :)
```
Comment (reyang) [inline build/RELEASING.md:80]: Do we really need these numbers? They will keep changing.

**Review by reyang — APPROVED** [review]
LGTM.

File: build/RELEASING.md (hunk 1 above)
Comment (reyang) [reply build/RELEASING.md:75]: The PR title has a typo which needs to be fixed.

File: build/RELEASING.md (hunk 3 above)
Comment (cijothomas) [reply build/RELEASING.md:80]: For now yes for easy referrals back and forth. I don't plan to do this manually for long :)
The exact steps, with step numbers are written down to make sure its easy for anyone to write automation for this.

**Pull request merged**